### PR TITLE
[SPARK-31638][WEBUI]Clean Pagination code for all webUI pages

### PR DESCRIPTION
**Pull request by iRakson**
### What changes were proposed in this pull request?

Pagination code across pages needs to be cleaned.
I have tried to clear out these things :
* Unused methods
* Unused method arguments
* remove redundant `if` expressions
* fix indentation

### Why are the changes needed?
This fix will make code more readable and remove unnecessary methods and variables.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Manually

### CODE REVIEW

**Comment by iRakson** [comment]
cc @srowen @dongjoon-hyun @HyukjinKwon Kindly Review

**Comment by srowen** [comment]
Jenkins test this please

**Comment by srowen** [comment]
Likewise I don't know this code well, but code cleanup is a good thing.
@sarutak or @zsxwing might know more.
I'm mostly trying to think if this can break anything, like replaying events from a history server with an old format or something. Not even sure that makes sense.

**Comment by sarutak** [comment]
I'll check within a week.

File: core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala [hunk 1]
```diff
@@ -523,13 +517,11 @@ private[ui] class JobPagedTable(
     store,
     data,
     basePath,
-    currentTime,
```
Comment (sarutak) [inline core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala:526]: If we can remove `currentTime` here, can we also remove `currentTime` from `JobPagedTable`?

File: core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala [hunk 2]
```diff
@@ -501,25 +499,21 @@ private[ui] class TaskPagedTable(
 
   override val dataSource: TaskDataSource = new TaskDataSource(
     stage,
-    currentTime,
```
Comment (sarutak) [inline core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala:504]: Similar to the comment for AllJobsPage, can we remove `currentTime` from `TaskPagedTable`?

File: core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala [hunk 3]
```diff
@@ -573,17 +563,9 @@ private[ui] class JobPagedTable(
...
-                  <span data-toggle="tooltip" data-placement="top" title={tooltip.get}>
-                    {header}&nbsp;{Unparsed(arrow)}
-                  </span>
-                } else {
-                  <span>
-                    {header}&nbsp;{Unparsed(arrow)}
-                  </span>
-                }
-              }
+              <span data-toggle="tooltip" data-placement="top" title={tooltip.getOrElse("")}>
```
Comment (sarutak) [inline core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala:566]: Nice catch. Tooltips with zero-length titles are never displayed.

File: core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala (hunk 1 above)
Comment (iRakson) [reply core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala:526]: yeah. It can be removed. Thanks.

**Comment by iRakson** [comment]
There is a bunch of code which is being copied whenever we implement pagination as pointed out [here](https://github.com/apache/spark/pull/28485#pullrequestreview-408881656).

I am thinking of making some changes to `PagedTable` implementation. Should I introduce them in another PR as this PR deals mostly with removal of dead code, indentation, etc. Also it already has huge amount of changes? @srowen @sarutak

**Comment by srowen** [comment]
Up to your judgment about whether they can be logically broken up into separate changes that are easier to reason about, or go together.

**Comment by iRakson** [comment]
@srowen @sarutak This PR has been updated as well.

**Comment by iRakson** [comment]
@srowen @sarutak PR has been updated. Kindly take a look at this one.

**Comment by iRakson** [comment]
retest this please

**Comment by sarutak** [comment]
retest this please.

**Comment by srowen** [comment]
Merged to master

**Comment by iRakson** [comment]
Thank You @srowen @sarutak

**Pull request closed without merging**